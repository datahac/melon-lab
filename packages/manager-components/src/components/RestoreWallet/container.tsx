import bip39 from 'bip39';
import { withFormik } from 'formik';
import { compose } from 'recompose';
import * as Yup from 'yup';
import RestoreWallet from './index';

const initialValues = {
  mnemonic: '',
  password: '',
};

const withFormValidation = withFormik({
  mapPropsToValues: props =>
    props.formValues ? { ...props.formValues } : initialValues,
  validationSchema: Yup.object().shape({
    mnemonic: Yup.string()
      .required('mnemonic is required.')
      .test('is-valid', 'Mnemonic is not valid', value =>
        bip39.validateMnemonic(value),
      ),
    password: Yup.string()
      .required('Password is required.')
      .min(8, 'Password needs to be at least 8 chars long. For your security!')
      .max(
        64,
        'Password is too long. Please reduce your password to max 64 chars!',
      ),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) =>
    form.props.onSubmit && form.props.onSubmit(values),
});

export default compose(withFormValidation)(RestoreWallet);
