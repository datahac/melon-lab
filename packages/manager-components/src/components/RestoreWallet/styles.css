.restore-wallet__input {
  margin-bottom: var(--space-m);
}

.restore-wallet__error {
  color: var(--color-loss);
}

.restore-wallet__actions {
  display: flex;
  justify-content: space-between;
}
