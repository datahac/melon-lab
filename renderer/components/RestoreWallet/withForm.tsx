import { withFormik } from 'formik';
import bip39 from 'bip39';
import * as Yup from 'yup';

export default withFormik({
  mapPropsToValues: () => ({
    mnemonic: '',
    password: '',
  }),
  validationSchema: Yup.object().shape({
    mnemonic: Yup.string()
      .required('mnemonic is required.')
      .test('is-valid', 'Mnemonic is not valid', value =>
        bip39.validateMnemonic(value),
      ),
    password: Yup.string()
      .required('Password is required.')
      .min(8, 'Password needs to be at least 8 chars long. For your security!')
      .max(
        64,
        'Password is too long. Please reduce your password to max 64 chars!',
      ),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) =>
    form.props.onSubmit && form.props.onSubmit(values),
});
