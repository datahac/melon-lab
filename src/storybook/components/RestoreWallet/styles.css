.restore-wallet {
  max-width: 768px;
  margin: 0 auto var(--space-xl) auto;
}

.restore-wallet__input {
  margin-bottom: var(--space-m);
}

.restore-wallet__error {
  color: var(--color-loss);
}

.restore-wallet__actions {
  display: flex;
  justify-content: space-between;
}
