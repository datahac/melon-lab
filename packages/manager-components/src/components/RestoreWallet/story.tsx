import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import RestoreWallet from './index';

const data = {
  values: {
    mnemonic: '',
    password: '',
  },
  touched: {
    mnemonic: null,
    password: null,
  },
  error: {
    mnemonic: null,
    password: null,
  },
  handleSubmit: action('onSubmit'),
};

storiesOf('Components|Restore Wallet', module).add('Default', () => {
  return <RestoreWallet {...data} />;
});
