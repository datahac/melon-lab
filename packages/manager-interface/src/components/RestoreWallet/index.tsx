import RestoreWallet from '~/components/RestoreWallet';
import { withRouter } from 'next/router';
import WalletMutation from './data/wallet';
import bip39 from 'bip39';
import { withFormik } from 'formik';
import { compose } from 'recompose';
import * as Yup from 'yup';

const initialValues = {
  mnemonic: '',
  password: '',
};

const withFormValidation = withFormik({
  mapPropsToValues: props =>
    props.formValues ? { ...props.formValues } : initialValues,
  validationSchema: Yup.object().shape({
    mnemonic: Yup.string()
      .required('mnemonic is required.')
      .test('is-valid', 'Mnemonic is not valid', value =>
        bip39.validateMnemonic(value),
      ),
    password: Yup.string()
      .required('Password is required.')
      .min(8, 'Password needs to be at least 8 chars long. For your security!')
      .max(
        64,
        'Password is too long. Please reduce your password to max 64 chars!',
      ),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) =>
    form.props.onSubmit && form.props.onSubmit(values),
});

const withRestoreWallet = BaseComponent => baseProps => (
  <WalletMutation onCompleted={() => {
    baseProps.router.replace({
      pathname: '/wallet',
    });
  }}>
    {(storeWallet, mutationProps) => (
      <BaseComponent
        onSubmit={values =>
          storeWallet({
            variables: { ...values },
          })
        }
        loading={mutationProps.loading}
      />
    )}
  </WalletMutation>
);

export default compose(
  withRouter,
  withRestoreWallet,
  withFormValidation,
)(RestoreWallet);
