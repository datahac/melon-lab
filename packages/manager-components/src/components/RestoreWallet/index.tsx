import React, { StatelessComponent } from 'react';
import Button from '~/blocks/Button';
import Form from '~/blocks/Form';
import Input from '~/blocks/Input';
import Spinner from '~/blocks/Spinner';
import Link from '~/blocks/Link';

import styles from './styles.css';

interface FormValues {
  mnemonic: string;
  password: string;
}

export interface RestoreWalletProps {
  errors?: any;
  handleBlur?: () => void;
  handleSubmit?: () => void;
  handleChange?: () => void;
  touched?: any;
  values: FormValues;
  loading?: boolean;
}

export const RestoreWallet: StatelessComponent<RestoreWalletProps> = ({
  errors,
  handleBlur,
  handleSubmit,
  handleChange,
  touched,
  values,
  loading,
}) => (
  <div className="restore-wallet">
    <style jsx>{styles}</style>
    {loading ? (
      <Spinner icon size="small" text="Restore Wallet..." />
    ) : (
      <Form onSubmit={handleSubmit}>
        <p>Please type your 12-words mnemonic:</p>
        <div className="restore-wallet__input">
          <Input
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.mnemonic}
            required={true}
            name="mnemonic"
            type="text"
            placeholder="mnemonic"
            error={touched.mnemonic && errors.mnemonic}
          />
        </div>
        <div className="restore-wallet__input">
          <Input
            maxlength={64}
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.password}
            required={true}
            name="password"
            type="password"
            placeholder="Set a Password"
            error={touched.password && errors.password}
          />
        </div>

        <div className="restore-wallet__actions">
          <div className="restore-wallet__action">
            <Link
              style="secondary"
              size="medium"
              href={{
                pathname: '/wallet',
              }}
            >
              Cancel
            </Link>
          </div>
          <div className="restore-wallet__action">
            <Button type="submit" style="primary">
              Restore
            </Button>
          </div>
        </div>
      </Form>
    )}
  </div>
);

export default RestoreWallet;
