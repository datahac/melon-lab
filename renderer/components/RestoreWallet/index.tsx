import React from 'react';
import RestoreWallet from '~/components/RestoreWallet';
import { withRouter } from 'next/router';
import WalletMutation from './data/wallet';
import withForm from './withForm';

const RestoreWalletForm = withForm(RestoreWallet);

const RestoreWalletContainer = ({ router }) => (
  <WalletMutation
    onCompleted={() => {
      router.push({
        pathname: '/wallet/overview',
      });
    }}
  >
    {(restoreWallet, restoreWalletProps) => {
      return (
        <RestoreWalletForm
          onSubmit={values => {
            restoreWallet({
              variables: {
                mnemonic: values.mnemonic,
                password: values.password,
              },
            });
          }}
          loading={restoreWalletProps.loading}
        />
      );
    }}
  </WalletMutation>
);

export default withRouter(RestoreWalletContainer);
