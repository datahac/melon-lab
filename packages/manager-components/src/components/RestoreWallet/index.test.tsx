import React from 'react';
import RestoreWallet from './container';

const data = {
  initialValues: {
    mnemonic: '',
  },
  onSubmit: () => null,
};

describe('RestoreWallet', () => {
  const defaultElement = <RestoreWallet {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
