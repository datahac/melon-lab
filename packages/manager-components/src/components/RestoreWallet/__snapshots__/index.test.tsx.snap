// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestoreWallet should render correctly 1`] = `
<Formik
  enableReinitialize={true}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "mnemonic": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={[Function]}
/>
`;
